packages/server/modules/gatekeeper/services/trialEmails.ts

```diff
@@ -0,0 +1,108 @@
+import { GetServerInfo } from '@/modules/core/domain/server/operations'
+import { FindEmailsByUserId } from '@/modules/core/domain/userEmails/operations'
+import {
+  EmailTemplateParams,
+  RenderEmail,
+  SendEmail,
+  SendEmailParams
+} from '@/modules/emails/domain/operations'
+import { getServerOrigin } from '@/modules/shared/helpers/envHelper'
+import { GetWorkspaceCollaborators } from '@/modules/workspaces/domain/operations'
+import { WorkspaceTeamMember } from '@/modules/workspaces/domain/types'
+import { Workspace } from '@/modules/workspacesCore/domain/types'
+import { Roles } from '@speckle/shared'
+
+type TrialExpiresArgs = {
+  workspace: Workspace
+  expiresInDays: number
+}
+
+type TrialExpiresArgsWithAdmin = TrialExpiresArgs & {
+  workspaceAdmin: WorkspaceTeamMember
+}
+
+const buildMjmlBody = ({
+  workspace,
+  expiresInDays,
+  workspaceAdmin
+}: TrialExpiresArgsWithAdmin) => {
+  const expireMessage =
+    expiresInDays === 0
+      ? `<strong>today</strong>!`
+      : `in <strong>${expiresInDays} days</strong>!`
+  const bodyStart = `<mj-text>
+Hello ${workspaceAdmin.name}!
+<br/>
+<br/>
+Your trial for your Speckle Workspace: <span style="font-variant: small-caps; font-weight: bold;">${workspace.name}</span> is about to expire ${expireMessage} 
+<br/>To keep using your workspace you have to upgrade to a paid plan.
+</mj-text>
+  `
+  const bodyEnd = `<mj-text>
+If you have any feedback/questions <a href="mailto:hello@speckle.systems" target="_blank">send us an email!</a>
+  </mj-text>`
+  return { bodyStart, bodyEnd }
+}
+
+const buildTextBody = () => {
+  const bodyStart = ''
+  return { bodyStart }
+}
+
+const buildEmailTemplateParams = (
+  args: TrialExpiresArgsWithAdmin
+): EmailTemplateParams => {
+  const url = new URL(`workspaces/${args.workspace.slug}`, getServerOrigin()).toString()
+  return {
+    mjml: buildMjmlBody(args),
+    text: buildTextBody(),
```

Comment:
What's up with this? It looks like `buildTextBody()` above doesn't return anything.

Reply:
it is fine
bodyStart and bodyEnd in emails are optional